src/main/java/com/ss/lms/entity/Branch.java

```diff
@@ -1,24 +1,66 @@
 package com.ss.lms.entity;
 
 import java.io.Serializable;
+import java.util.List;
 
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinTable;
+import javax.persistence.ManyToMany;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+
+import org.hibernate.annotations.Where;
 import org.springframework.stereotype.Component;
 
+import com.fasterxml.jackson.annotation.JsonIdentityInfo;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.ObjectIdGenerators;
+
 @Component
+@Entity
+@Table(name = "tbl_library_branch")
+@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "branchId")
 public class Branch implements Serializable {
+	
 	/**
 	 * 
 	 */
-	private static final long serialVersionUID = 1118815487251491971L;
-	private Integer branchId;
+	private static final long serialVersionUID = 7877691824229470895L;
+
+	@Id
+	@Column(name="branchId", nullable = false)
```

Comment:
doesn't this take affect during serialization too?? You wouldnt be able to Post a new branch with "nullable = false;". Unless im misinterpreting how that works.